pkg/status/render.go

```diff
@@ -56,29 +56,53 @@ func FormatStatus(data []byte) (string, error) {
-		renderStatusTemplate(b, "/systemprobe.tmpl", systemProbeStats)
+
+	headerFunc := func() { renderStatusTemplate(b, "/header.tmpl", stats) }
+	checkStatsFunc := func() {
+		renderChecksStats(b, runnerStats, pyLoaderStats, pythonInit, autoConfigStats, checkSchedulerStats, inventoriesStats, "")
+	}
+	jmxFetchFunc := func() { renderStatusTemplate(b, "/jmxfetch.tmpl", stats) }
+	forwarderFunc := func() { renderStatusTemplate(b, "/forwarder.tmpl", forwarderStats) }
+	endpointsFunc := func() { renderStatusTemplate(b, "/endpoints.tmpl", endpointsInfos) }
+	logsAgentFunc := func() { renderStatusTemplate(b, "/logsagent.tmpl", logsStats) }
+	systemProbeFunc := func() {
+		if config.Datadog.GetBool("system_probe_config.enabled") {
+			renderStatusTemplate(b, "/systemprobe.tmpl", systemProbeStats)
+		}
+	}
+	processAgentFunc := func() { renderStatusTemplate(b, "/process-agent.tmpl", processAgentStatus) }
+	traceAgentFunc := func() { renderStatusTemplate(b, "/trace-agent.tmpl", stats["apmStats"]) }
+	aggregatorFunc := func() { renderStatusTemplate(b, "/aggregator.tmpl", aggregatorStats) }
+	dogstatsdFunc := func() { renderStatusTemplate(b, "/dogstatsd.tmpl", dogstatsdStats) }
+	clusterAgentFunc := func() {
+		if config.Datadog.GetBool("cluster_agent.enabled") || config.Datadog.GetBool("cluster_checks.enabled") {
+			renderStatusTemplate(b, "/clusteragent.tmpl", dcaStats)
+		}
 	}
-	renderStatusTemplate(b, "/process-agent.tmpl", processAgentStatus)
-	renderStatusTemplate(b, "/trace-agent.tmpl", stats["apmStats"])
-	renderStatusTemplate(b, "/aggregator.tmpl", aggregatorStats)
-	renderStatusTemplate(b, "/dogstatsd.tmpl", dogstatsdStats)
-	if config.Datadog.GetBool("cluster_agent.enabled") || config.Datadog.GetBool("cluster_checks.enabled") {
-		renderStatusTemplate(b, "/clusteragent.tmpl", dcaStats)
+	snmpTrapFunc := func() {
+		if traps.IsEnabled() {
+			renderStatusTemplate(b, "/snmp-traps.tmpl", snmpTrapsStats)
+		}
 	}
-	if traps.IsEnabled() {
-		renderStatusTemplate(b, "/snmp-traps.tmpl", snmpTrapsStats)
+	autodiscoveryFunc := func() {
+		if config.IsContainerized() {
+			renderAutodiscoveryStats(b, stats["adEnabledFeatures"], stats["adConfigErrors"], stats["filterErrors"])
+		}
 	}
-	if config.IsContainerized() {
-		renderAutodiscoveryStats(b, stats["adEnabledFeatures"], stats["adConfigErrors"], stats["filterErrors"])
+
+	// Sections of the status output that should be present in either the CLC or regular agent
+	renderFuncsByType := map[string][]func(){
+		"clc": {headerFunc, checkStatsFunc, aggregatorFunc, endpointsFunc, clusterAgentFunc, autodiscoveryFunc},
+		"agent": {headerFunc, checkStatsFunc, jmxFetchFunc, forwarderFunc, endpointsFunc, logsAgentFunc, systemProbeFunc,
+			processAgentFunc, traceAgentFunc, aggregatorFunc, dogstatsdFunc, clusterAgentFunc, snmpTrapFunc, autodiscoveryFunc}}
+
+	if config.IsCLCRunner() {
+		renderAgentSections("clc", renderFuncsByType)
+		return b.String(), nil
 	}
 
+	renderAgentSections("agent", renderFuncsByType)
+
```

Comment:
maybe the `map` here is not needed we can image to replace it by

```suggestion
    var renderFuncs []func()
	if config.IsCLCRunner() {
	    renderFuncs = []func(){headerFunc, checkStatsFunc, aggregatorFunc, endpointsFunc, clusterAgentFunc, autodiscoveryFunc}
    } else {
      renderFuncs = []func(){headerFunc, checkStatsFunc, jmxFetchFunc, forwarderFunc, endpointsFunc, logsAgentFunc, systemProbeFunc,
			processAgentFunc, traceAgentFunc, aggregatorFunc, dogstatsdFunc, clusterAgentFunc, snmpTrapFunc, autodiscoveryFunc}}
    }
    renderAgentSections(renderFuncs)

```

Reply:
Updated with these changes!